nndownload/nndownload.py

```diff
@@ -1267,6 +1268,28 @@ def perform_ffmpeg_dl(video_id: AnyStr, filename: AnyStr, duration: float, strea
         raise FormatNotAvailableException("Failed to download video or audio stream")
 
 
+def perform_native_hls_dl(session: requests.Session, filename: AnyStr, duration: float, m3u8_streams: List, threads: int = 1):
+    """Download video and audio streams using native HLS downloader and merge using ffmpeg."""
+
+    for stream, stream_filename in m3u8_streams:
+        download_hls(stream, stream_filename, session=session, threads=threads)
+
+    if len(m3u8_streams) > 1:
+        video_convert = FfmpegDL(streams=[stream_filename for _, stream_filename in m3u8_streams],
+                                 input_kwargs={},
+                                 output_path=filename,
+                                 output_kwargs={
+                                     "vcodec": "copy",
+                                     "acodec": "copy",
+                                 })
+        video_convert.convert(name='Merging audio and video', duration=duration)
+        for _, stream_filename in m3u8_streams:
+            os.remove(stream_filename)
+    else:
+        os.rename(m3u8_streams[0][1], filename)
```

Comment:
Coupled with my earlier comment, I think it'd be better to write to the temp dir and then move as needed.